### ISSUE TITLE
Schema changes (`index_columns`) when doing a merge

### ISSUE THREAD

**Issue by JSpenced** [lancedb/lance#1213]
This might be the desired behavior and I am not sure the `index_columns` do anything special except when converting back into pandas? Like in the back end it doesn't matter if you set `index_columns` because they aren't special?

But that being said before and after a merge it doesn't preserve the `index_column`. The merge is done on the `index_column` but this is probably more the normal behavior if an index is set.

Before merge:
```python
inference_date: timestamp[ns]
source_parent_id: int16
source_parent: string
source_count_distinct: int32
merge_key: string
-- schema metadata --
pandas: '{"index_columns": ["merge_key"], "column_indexes": [{"name": nul' + 916
```

After merge:
```python
inference_date: timestamp[ns]
source_parent_id: int16
source_parent: string
source_count_distinct: int32
merge_key: string
assignment_pair: int32
__index_level_0__: int64
-- schema metadata --
pandas: '{"index_columns": ["__index_level_0__"], "column_indexes": [{"na' + 602
```

**Comment by wjones127** [comment]
> This might be the desired behavior and I am not sure the index_columns do anything special except when converting back into pandas? Like in the back end it doesn't matter if you set index_columns because they aren't special and you don't get any speedup on filters, for example?

You are correct that Lance does nothing special with the pandas index columns. They just become a normal column in the dataset.

The new metadata after the merge, does that by any chance reflect the index columns of the right-hand side?

**Comment by JSpenced** [comment]
Yes, it does. Good call! Normal pandas won't allow merges on index to columns I didn't notice that.

If I set the `merge_key` as index then it has the correct schema. Thanks!

**Comment by wjones127** [comment]
Thanks for confirming. I think the fix we want is that `merge()` should merge the schema metadata, but give the existing schema metadata precedence.

**Comment by JSpenced** [comment]
I would agree it should take the left side over the right, so thanks!
